srcpkgs/xbps-triggers/files/initramfs-regenerate

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+#
+# Trigger to regenerate an initramfs for every kernel with a module directory
+# in lib/modules on package post-install.
+#
+# To change the initramfs generator, edit or create the file
+# etc/default/initramfs-regenerate and add or change the line
+#
+#     INITRAMFS_GENERATOR=<generator>
+#
+# where <generator> is one of "dracut", "mkinitcpio" or "none". By default, a
+# value of "dracut" is assumed.
+#
+# Arguments:	$ACTION = [run/targets]
+#		$TARGET = [post-install/post-remove]
+#		$PKGNAME
+#		$VERSION
+#		$UPDATE = [yes/no]
+
+ACTION="$1"
+TARGET="$2"
+PKGNAME="$3"
+VERSION="$4"
+UPDATE="$5"
+
+export PATH="usr/bin:/usr/sbin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+case "$ACTION" in
+	targets)
+		echo "post-install post-remove"
+		exit 0
+		;;
+	run)
+		;;
+	*)
+		exit 1
+		;;
+esac
+
+# Read the configuration, if it exists
+[ -f etc/default/initramfs-regenerate ] && . etc/default/initramfs-regenerate
+
+case "${INITRAMFS_GENERATOR:-dracut}" in
+	dracut)
+		if command -v dracut >/dev/null 2>&1; then
+			echo "Regenerating initramfs with dracut"
+			dracut -f -q --regenerate-all
+		fi
+		;;
+	mkinitcpio)
+		if command -v mkinitcpio >dev/null 2>&1; then
+			echo "Regenerating initramfs with mkinitcpio"
+			# Need to manually mimic dracut's --regenerate-all
+			for kdir in lib/modules/*; do
+				[ -f "${kdir}/modules.dep" ] || [ -f "${kdir}/modules/dep.bin" ] || continue
```

Comment:
I assume this is the check that dracut runs? It has issues with our kernel erasure and leads to dracut trying to generate initramfs for old deleted kernels.

I thought @Duncaen had documented the issue with the kernel packages and how they work, but I can't find it rn.

Reply:
This just uses the same logic that `dracut --regenerate-all` uses to iterate over kernels.

Reply:
If you can think of a better check, that would make the mkinitcpio experience a tad better than the dracut one, specifically on Void.

Reply:
I switched to `[ -d "${kdir}/kernel" ]`, since that directory contains the actual modules and its existence seems better tied to the lifetime of installed kernels.